### ISSUE TITLE
Nextcloud doesn't run on KDE Neon

### ISSUE THREAD

**Issue by twosky2000** [kyrofa/nextcloud-snap#1]
Expected behavior:
- sudo snap install nextcloud
- Go to the Webinterface (127.0.0.1, localhost, [Name].local)

Actual behavior:
- sudo snap install nextcloud
- Nextcloud not reachable, netstat gives no clues

I'm not familiar with snap, so i don't know how to debug them, or where things shows up.
I'm running Ubuntu Neon

I will try to get some Wiki entries to debug the snap, when something like this happens.

**Comment by kyrofa** [comment]
Hey @twosky2000, thanks for giving the snap a test!

Can you please provide the output of the following commands?

`$ sudo systemctl status snap.nextcloud.apache`
`$ sudo systemctl status snap.nextcloud.mysql`

**Comment by twosky2000** [comment]
services aren't runninging
Apache:
> ● snap.nextcloud.apache.service - Service for snap application nextcloud.apache
>    Loaded: loaded (/etc/systemd/system/snap.nextcloud.apache.service; enabled; vendor preset: enabled)
>    Active: inactive (dead) (Result: exit-code) since Don 2016-06-16 12:11:24 CEST; 2h 36min ago
>   Process: 1112 ExecStop=/usr/bin/ubuntu-core-launcher snap.nextcloud.apache snap.nextcloud.apache /snap/nextcloud/1/stop-command-apache.wrapper (code=exited, status=1/FAILURE)
>   Process: 1109 ExecStart=/usr/bin/ubuntu-core-launcher snap.nextcloud.apache snap.nextcloud.apache /snap/nextcloud/1/command-apache.wrapper (code=exited, status=1/FAILURE)
>  Main PID: 1109 (code=exited, status=1/FAILURE)
> 
> Jun 16 12:11:24 RocketOne systemd[1]: snap.nextcloud.apache.service: Unit entered failed state.
> Jun 16 12:11:24 RocketOne systemd[1]: snap.nextcloud.apache.service: Failed with result 'exit-code'.
> Jun 16 12:11:24 RocketOne systemd[1]: snap.nextcloud.apache.service: Service hold-off time over, scheduling restart.
> Jun 16 12:11:24 RocketOne systemd[1]: Stopped Service for snap application nextcloud.apache.
> Jun 16 12:11:24 RocketOne systemd[1]: snap.nextcloud.apache.service: Start request repeated too quickly.
> Jun 16 12:11:24 RocketOne systemd[1]: Failed to start Service for snap application nextcloud.apache.


mysql:

> ● snap.nextcloud.mysql.service - Service for snap application nextcloud.mysql
>    Loaded: loaded (/etc/systemd/system/snap.nextcloud.mysql.service; enabled; vendor preset: enabled)
>    Active: inactive (dead) (Result: exit-code) since Don 2016-06-16 14:45:13 CEST; 2min 38s ago
>   Process: 9256 ExecStop=/usr/bin/ubuntu-core-launcher snap.nextcloud.mysql snap.nextcloud.mysql /snap/nextcloud/1/stop-command-mysql.wrapper (code=exited, status=1/FAILURE)
>   Process: 9253 ExecStart=/usr/bin/ubuntu-core-launcher snap.nextcloud.mysql snap.nextcloud.mysql /snap/nextcloud/1/command-mysql.wrapper (code=exited, status=1/FAILURE)
>  Main PID: 9253 (code=exited, status=1/FAILURE)
> 
> Jun 16 14:45:13 RocketOne systemd[1]: snap.nextcloud.mysql.service: Unit entered failed state.
> Jun 16 14:45:13 RocketOne systemd[1]: snap.nextcloud.mysql.service: Failed with result 'exit-code'.
> Jun 16 14:45:13 RocketOne systemd[1]: snap.nextcloud.mysql.service: Service hold-off time over, scheduling restart.
> Jun 16 14:45:13 RocketOne systemd[1]: Stopped Service for snap application nextcloud.mysql.
> Jun 16 14:45:13 RocketOne systemd[1]: snap.nextcloud.mysql.service: Start request repeated too quickly.
> Jun 16 14:45:13 RocketOne systemd[1]: Failed to start Service for snap application nextcloud.mysql.

**Comment by kyrofa** [comment]
Huh, odd that they're both down. Okay, let's focus on Apache first. As a sanity check: you don't already have something running on port 80, do you? Assuming not, let's continue.

Start Apache back up again:

`$ sudo systemctl start snap.nextcloud.apache`

Now take a look at the syslog. You'll see Apache try to restart a number of times before finally giving up. Do you see a reason for the failure? Please paste the resulting log in here.

**Comment by twosky2000** [comment]
I have nothing at port 80. System was just set up.
i tried to restart

Here's what i get from journalctl:
```
-- Logs begin at Don 2016-06-16 14:55:05 CEST, end at Don 2016-06-16 15:04:55 CEST. --
Jun 16 14:55:06 RocketOne systemd[1]: Started Service for snap application nextcloud.apache.
Jun 16 14:55:06 RocketOne ubuntu-core-launcher[827]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Main process exited, code=exited, status=1/FAILURE
Jun 16 14:55:06 RocketOne ubuntu-core-launcher[967]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Control process exited, code=exited status=1
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Unit entered failed state.
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Failed with result 'exit-code'.
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Service hold-off time over, scheduling restart.
Jun 16 14:55:06 RocketOne systemd[1]: Stopped Service for snap application nextcloud.apache.
Jun 16 14:55:06 RocketOne systemd[1]: Started Service for snap application nextcloud.apache.
Jun 16 14:55:06 RocketOne ubuntu-core-launcher[994]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Main process exited, code=exited, status=1/FAILURE
Jun 16 14:55:06 RocketOne ubuntu-core-launcher[998]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Control process exited, code=exited status=1
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Unit entered failed state.
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Failed with result 'exit-code'.
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Service hold-off time over, scheduling restart.
Jun 16 14:55:06 RocketOne systemd[1]: Stopped Service for snap application nextcloud.apache.
Jun 16 14:55:06 RocketOne systemd[1]: Started Service for snap application nextcloud.apache.
Jun 16 14:55:06 RocketOne ubuntu-core-launcher[1035]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Main process exited, code=exited, status=1/FAILURE
Jun 16 14:55:06 RocketOne ubuntu-core-launcher[1043]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Control process exited, code=exited status=1
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Unit entered failed state.
Jun 16 14:55:06 RocketOne systemd[1]: snap.nextcloud.apache.service: Failed with result 'exit-code'.
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Service hold-off time over, scheduling restart.
Jun 16 14:55:07 RocketOne systemd[1]: Stopped Service for snap application nextcloud.apache.
Jun 16 14:55:07 RocketOne systemd[1]: Started Service for snap application nextcloud.apache.
Jun 16 14:55:07 RocketOne ubuntu-core-launcher[1052]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Main process exited, code=exited, status=1/FAILURE
Jun 16 14:55:07 RocketOne ubuntu-core-launcher[1063]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Control process exited, code=exited status=1
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Unit entered failed state.
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Failed with result 'exit-code'.
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Service hold-off time over, scheduling restart.
Jun 16 14:55:07 RocketOne systemd[1]: Stopped Service for snap application nextcloud.apache.
Jun 16 14:55:07 RocketOne systemd[1]: Started Service for snap application nextcloud.apache.
Jun 16 14:55:07 RocketOne ubuntu-core-launcher[1069]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Main process exited, code=exited, status=1/FAILURE
Jun 16 14:55:07 RocketOne ubuntu-core-launcher[1078]: aa_change_onexec failed with -1. errmsg: No such file or directory
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Control process exited, code=exited status=1
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Unit entered failed state.
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Failed with result 'exit-code'.
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Service hold-off time over, scheduling restart.
Jun 16 14:55:07 RocketOne systemd[1]: Stopped Service for snap application nextcloud.apache.
Jun 16 14:55:07 RocketOne systemd[1]: snap.nextcloud.apache.service: Start request repeated too quickly.
Jun 16 14:55:07 RocketOne systemd[1]: Failed to start Service for snap application nextcloud.apache.
```

**Comment by kyrofa** [comment]
apparmor issues! That looks like a snapd bug-- investigating now, thank you very much for reporting!

**Comment by zyga** [comment]
What is the distribution that was hosting snapd when that error occurred?

**Comment by kyrofa** [comment]
From the description, sounds like Ubuntu Neon.

**Comment by zyga** [comment]
Can we please see the `/etc/os-release` file from that system?

**Comment by twosky2000** [comment]
Here's /etc/os-release:
NAME="KDE neon"
VERSION="5.6"
ID=neon
ID_LIKE=ubuntu debian
PRETTY_NAME="KDE neon User Edition 5.6"
VERSION_ID="16.04"
HOME_URL="http://neon.kde.org/"
SUPPORT_URL="http://neon.kde.org/"
BUG_REPORT_URL="http://bugs.kde.org/"
UBUNTU_CODENAME=xenial

**Comment by zyga** [comment]
I think this is the problem

What is the packaging of snap-confine (aka ubuntu-core-launcher) for this distribution?
Is this distribution using an Ubuntu kernel?

**Comment by zyga** [comment]
Please patch this file and send the pull request along with a trivial test:

https://github.com/snapcore/snapd/blob/master/release/release.go#L54

**Comment by kyrofa** [comment]
Alright, I'll take a look at that.

**Comment by twosky2000** [comment]
Not sure how to do that. 
I tried sudo snap refresh ubuntu-core
```
error: cannot perform the following tasks:
- Download snap "ubuntu-core" from channel "stable" (revision 122 of snap "ubuntu-core" already installed)
```
Will research more to get the patch on my system.

**Comment by kyrofa** [comment]
@twosky2000 if you want to hack on snapd please feel free, but I'm getting setup with a Neon VM now to play with this issue.

**Comment by kyrofa** [comment]
@twosky2000 @zyga verified: it's a bug, apparmor profiles aren't generated on neon. Making a PR now. Thanks again @twosky2000!

**Comment by kyrofa** [comment]
Fixed in snapcore/snapd#1346.

**Comment by kyrofa** [comment]
Note that the fix won't make it into the archives until next week, but if you want to continue playing with the snap you can install snapd from this PPA: https://launchpad.net/~snappy-dev/+archive/ubuntu/edge
